lib/chef/provider/package/homebrew.rb

```diff
@@ -24,31 +24,40 @@ class Chef
   class Provider
     class Package
       class Homebrew < Chef::Provider::Package
+        allow_nils
+        use_multipackage_api
 
         provides :package, os: "darwin", override: true
         provides :homebrew_package
 
         include Chef::Mixin::HomebrewUser
 
         def load_current_resource
-          self.current_resource = Chef::Resource::HomebrewPackage.new(new_resource.name)
+          @current_resource = Chef::Resource::HomebrewPackage.new(new_resource.name)
           current_resource.package_name(new_resource.package_name)
-          current_resource.version(current_installed_version)
+          current_resource.version(get_current_versions)
           logger.trace("#{new_resource} current version is #{current_resource.version}") if current_resource.version
 
-          @candidate_version = candidate_version
-
-          logger.trace("#{new_resource} candidate version is #{@candidate_version}") if @candidate_version
-
           current_resource
         end
 
-        def install_package(name, version)
-          unless current_resource.version == version
-            brew("install", options, name)
+        def candidate_version
+          package_name_array.map do |package_name|
+            available_version(package_name)
           end
         end
 
+        def get_current_versions
+          package_name_array.map do |package_name|
+            installed_version(package_name)
+          end
+        end
+
+        def install_package(names, versions)
+          packages = names.select { |x| x unless x.nil? }
```

Comment:
If this is getting rid of potential nils, `compact` does the same thing:

```
irb(main):001:0> a = [1, nil, 2, nil, 3]
=> [1, nil, 2, nil, 3]
irb(main):002:0> a.select { |x| x unless x.nil? }
=> [1, 2, 3]
irb(main):003:0> a.compact
=> [1, 2, 3]
```

Reply:
Updates to use compact. Thanks